### ISSUE TITLE
Error while calling the api

### ISSUE THREAD

**Issue by zilberd** [microsoftgraph/microsoft-graph-docs#8103]
Hi,

I am trying to onboard an AD group via the graph api.
My user is user admin on the directory, owning the group and also owning the target catalog.

payload is 
{
  "requestType": "AdminAdd",
  "isValidationOnly": true,
  "catalogId": "304d2042-1159-4ff5-b3bc-7f5bf22633a5",
  "accessPackageResource": {
    "originSystem": "AadGroup",
    "originId": "393a2d2b-48f5-438a-a9c6-57d514b0606c"
  }
}

I am getting an unknown error 
{
  "error": {
    "code": "UnknownError",
    "message": "",
    "innerError": {
      "request-id": "7ff7b3d6-b890-4ee2-9425-0f2337868b71",
      "date": "2020-05-07T13:45:50"
    }
  }
}

I use the list api call with no trouble

screenshot when workgin from the portal
request:
![image](https://user-images.githubusercontent.com/5602047/81376918-b7eed200-9104-11ea-9844-1ce5549c400e.png)
working change:
![image](https://user-images.githubusercontent.com/5602047/81376996-dead0880-9104-11ea-804a-0f4f678ac357.png)

then done via graph api
request (same values)
![image](https://user-images.githubusercontent.com/5602047/81377078-0c924d00-9105-11ea-8592-7a7dc1f1f8a5.png)

non working action, not all the values were passed to it if you compare to the portal action and you can see the target is all 0000
![image](https://user-images.githubusercontent.com/5602047/81377198-3fd4dc00-9105-11ea-969f-6bd934f54077.png)

thanks
Dimitri

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ec50888f-1fe5-5d73-3a4e-cfd698d6b455
* Version Independent ID: 4786d6a3-0108-20a2-d0fc-b1e727682d57
* Content: [Create accessPackageResourceRequest - Microsoft Graph beta](https://docs.microsoft.com/en-us/graph/api/accesspackageresourcerequest-post?view=graph-rest-beta&tabs=http#feedback)
* Content Source: [api-reference/beta/api/accesspackageresourcerequest-post.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/beta/api/accesspackageresourcerequest-post.md)
* Product: **microsoft-identity-platform**
* Technology: **microsoft-graph**
* GitHub Login: @markwahl-msft
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by markwahl-msft** [comment]
There is a known limitation with the API adding a resource to a catalog. At this time it is only possible for a global administrator to create a new access package resource request.

**Comment by markwahl-msft** [comment]
@zilberd we're deploying an update for this issue.  Could you retry the operation?  Thanks, Mark

**Comment by zilberd** [comment]
@markwahl-msft it looks better but it does require User Admin role as minimum for it to work

I have tested the following:
group already onboarded to the catalog
- elevated as user admin: working
- elevated as group admin: not working
- not elevated: not working

group not onboarded to the catalog
- elevated as user admin: not working
- elevated as group admin: not working
- not elevated: not working
Not sure for this last one what is the correct call for onboarding a group to the catalog.


over the portal when non elevated my user can add the group to the application package (and onboard the group to the catalog as he is owning it)

**Comment by markwahl-msft** [comment]
The group admin role is not currently used in Entitlement Management, see https://docs.microsoft.com/en-us/azure/active-directory/governance/entitlement-management-delegate  

However, a user who is both an owner of group and an owner of catalog should be able to add that group to that catalog using the https://docs.microsoft.com/en-us/graph/api/accesspackageresourcerequest-post?view=graph-rest-beta&tabs=http API, without needing to be in an administrative role.  I noticed that in your example at the beginning of the thread, you had "isValidationOnly": true, you do not want to include that in the payload since that causes the resource to not be added - that is an authZ check only.  FYI there is also a new tutorial that has an example of how to add a group to a catalog, and add a group role to an access package, at https://docs.microsoft.com/en-us/graph/tutorial-access-package-api

**Comment by zilberd** [comment]
sorry I was not specific enough.
I did the test with the group admin in order to be sure that I was hitting any group ownership issue.

I can confirm that it is only working with an elevated user.
Even though my user if owning the catalog, and the group he is getting this error when trying to add the group.
{
  "error": {
    "code": "UnAuthorized",
    "message": "User is not authorized to perform the operation. Reason: Unauthorized",
    "innerError": {
      "date": "2020-09-11T12:50:42",
      "request-id": "27a4fa48-4819-400a-b26d-274c67a443e8"
    }
  }
}

this is working well when my user is elevated as User Admin (we are using PIM).

Thanks for the tip about the tutorial.
I followed it just to check and I am getting the same error as befor when trying to import the group in the catalog:
{
  "error": {
    "code": "InvalidModel",
    "message": "Model is not valid.",
    "innerError": {
      "date": "2020-09-11T12:50:21",
      "request-id": "2b527311-fb6d-4ea5-a912-39ae3330777f"
    }
  }
}

I will open a comment there as well.

BTW, I forgot to say that all the calls are made over powershell, call a dedicated app registered in AAD.
The api itself has no role in AAD and in graph API, it only has Directory.AccessAsUser.All as delegate right
